ui/js/services/perf/compare.js

```diff
@@ -217,62 +217,52 @@ treeherder.factory('PhCompare', [
+                                        } else {
+                                            entry[signatureHash].values.push(datum.value);
+                                        }
+                                    });
                                 }
-                                _.forIn(seriesData, function (data, signature) {
-                                    // Aggregates data from the server on a single group of values which
-                                    // will be compared later to another group. Ends up with an object
-                                    // with description (name/platform) and values.
-                                    // The values are later processed at getCounterMap as the data arguments.
-                                    var values = data
-                                        .filter(datum => datum.push_id === resultSetId)
-                                        .map(datum => datum.value);
-                                    var seriesData = _.find(seriesList, { signature: signature });
-                                    if (seriesData) {
-                                        resultsMap[resultSetId][signature] = {
-                                            platform: seriesData.platform,
-                                            name: seriesData.name,
-                                            lowerIsBetter: seriesData.lowerIsBetter,
-                                            frameworkId: seriesData.frameworkId,
-                                            values: values
-                                        };
-                                    }
-                                });
                             });
-                        });
-                    }
-                    return PhSeries.getSeriesData(
-                            projectName, {
-                                signatures: _.map(seriesChunk, 'signature'),
-                                framework: _.uniq(_.map(seriesChunk, 'frameworkId')),
-                                interval: params.interval.value }
-                        ).then(function (seriesData) {
-                            _.forIn(seriesData, function (data, signature) {
-                                var series = _.find(seriesChunk, { signature: signature });
-                                resultsMap[signature] = {
-                                    platform: series.platform,
-                                    name: series.name,
-                                    lowerIsBetter: series.lowerIsBetter,
-                                    frameworkId: series.frameworkId,
-                                    values: _.map(data, 'value')
-                                };
-                            });
-                        });
-                })).then(function () {
-                    return resultsMap;
-                });
+                        })
+                )).then(() => resultsMap);
```

Comment:
Oh cool so you don't even need a 'return' before resultsMap (?)

Reply:
Nope! Return is assumed if the function is a single expression. :)